OpenEdX/Managers/DeepLinkManager/DeepLinkManager.swift

```diff
@@ -0,0 +1,71 @@
+//
+//  DeepLinkManager.swift
+//  OpenEdX
+//
+//  Created by Anton Yarmolenka on 24/01/2024.
+//
+
+import Foundation
+import Core
+import UIKit
+
+public protocol DeepLinkService {
+    func configureWith(launchOptions: [UIApplication.LaunchOptionsKey: Any]?)
+    func handledURLWith(app: UIApplication, open url: URL, options: [UIApplication.OpenURLOptionsKey: Any]) -> Bool
+}
+
+class DeepLinkManager {
+    private var service: DeepLinkService?
+    
+    // Init manager
+    public init(config: ConfigProtocol) {
+        self.service = self.serviceFor(config: config)
+    }
+    
+    private func serviceFor(config: ConfigProtocol) -> DeepLinkService? {
+        // init deep link service
+        if config.branch.enabled {
+            return BranchService()
+        }
+        return nil
+    }
+    
+    // check if service is added (means enabled)
+    var serviceEnabled: Bool {
+        service != nil
+    }
+    
+    // Configure services
+    func configureDeepLinkService(launchOptions: [UIApplication.LaunchOptionsKey: Any]?) {
+        if let service = service {
+            service.configureWith(launchOptions: launchOptions)
+        }
+    }
+    
+    // Handle open url
+    func handledURLWith(
+        app: UIApplication,
+        open url: URL,
+        options: [UIApplication.OpenURLOptionsKey: Any] = [:]
+    ) -> Bool {
+        if let service = service {
+            return service.handledURLWith(app: app, open: url, options: options)
+        }
+        return false
+    }
+    
+    // This method process push notification with the link object
```

Comment:
We might need to add a more descriptive comment here because it's a bit odd to have push notification process in the deeplink manager. Or maybe we can create another class that will handle data that received from push and deep link and router the learner to the desired screen.

Reply:
I will change comment. DeepLinkManager can get links from deep link service and also link can be extracted from push notifications body. So we have now two functions to process these links: 
`func processDeepLink(with params: [String: Any]) {`
and
`func processLinkFromNotification(_ link: PushLink) {`
both functions should try to detect redirection type and parameters and call redirection process